authentication/src/main/java/io/camunda/authentication/filters/TenantRequestAttributeFilter.java

```diff
@@ -56,24 +55,17 @@ protected void doFilterInternal(
     if (!multiTenancyCfg.isEnabled()) {
       return Set.of(TenantOwned.DEFAULT_TENANT_IDENTIFIER);
     }
-
-    return switch (principal) {
-      case final UsernamePasswordAuthenticationToken token -> {
-        if (!(token.getPrincipal() instanceof final CamundaUser user)) {
-          LOG.error("cannot find tenants: principal is not a camunda user");
-          yield null;
-        }
-        yield user.getTenants().stream().map(TenantDTO::tenantId).collect(Collectors.toSet());
-      }
-      case final OAuth2AuthenticationToken token -> {
-        LOG.error("cannot find tenants: tenant mapping isn't implemented for oidc");
-        yield null;
-      }
-      default -> {
-        LOG.error(
-            "cannot find tenants: unsupported principal type {}", principal.getClass().getName());
-        yield null;
-      }
-    };
+    if (!(principal instanceof final Authentication auth)) {
+      LOG.error(
+          "cannot find tenants: unsupported principal type: {}", principal.getClass().getName());
+      return null;
+    }
+    if (!(auth.getPrincipal() instanceof final CamundaPrincipal camundaPrincipal)) {
```

Comment:
❓ When testing with a client (not user) and enabled tenancy, any request fails with 

```
ERROR io.camunda.authentication.filters.TenantRequestAttributeFilter - cannot find tenants: unsupported principal type: org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
```

Will the client case be handled separately?